typeinfo-less.md

```diff
@@ -0,0 +1,362 @@
+following manner:
+
+Qualifiers are each assigned a score
+
+> &: 1
+> &&: 2
+> const: 3
+> volatile: 6
+
+and ordering lowest-first after summing them.
+
+For any unqualified type `T`, the order of all possible qualified types would
+be:
+
+```cpp
+0  T
+1  T &
+2  T &&
+3  T const
+4  T const &
+5  T const &&
+6  T volatile
+7  T volatile &
+8  T volatile &&
+9  T const volatile
+10 T const volatile &
+11 T const volatile &&
+```
+
+The remainder of the paper concerns itself only with unqualified types.
+
+### Ordering Native Types
+
+We order scalar types before any compound types; bult-in types first, followed 
+by user-defined types.
+
+built-in types with simple names must be ordered before any types that reference
+other types.
+
+In particular, scalar types should be ordered as follows:
+
+1. `void` comes first because it's not reifiable,
+2. `nullptr_t` as the first monostate
+3. (any other monostates, if we ever add them)
+4. `bool` as the first bi-state
+5. (any other bi-states, if we ever add them)
+6. Raw-memory types (`char`, `signed char`, `unsigned char`, std::byte)
+7. Integral types in order of size, signed before unsigned (`short`, `unsigned short`, `int`, `unsigned int`, `long`, `unsigned long`, `long long`, `unsigned long long`, followed by any implementation-defined wider integral types like __int128_t etc.). Intersperse any implementation-defined built-in integral types as needed between the above.
+8. Any remaining character types that are not type-aliases of any of the above, including unicode, according to the following rules: smallest first, unicode-specific variants after non-unicode variants.
+9. Floating-point types, in order of size; tie-breaks go in favor of `float`, `double` and `long double` (so types from https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p1467r9.html are after same-sized "regular" floating-point types)
+10. Enumeration types (internally ordered by rules for class type ordering by name)
+11. Function types (internally ordered by rules in section [function-types])
+12. Pointer types (internally ordered by their pointee-type)
+13. Pointer-to-member types (internally ordered by pointee-type)
+
+### Ordering Compound types:
+
+Array types will be ordered after scalar types but before class types. 
+
+Order arrays first internally by element type, then by rank, then by rank
+bounds, lowest first.
+
+So the order of the following, for a given type T:
+
+```cpp
+T[10]
+T[11]
+T[10][2]
+T[3][2]
+```
+
+would be ordered `T[10] < T[11] < T[3][2] < T[10][2]`
+
+Class types will be ordered according to the rules below, see [ordering-classes]
+
+### namespaces
+
+Namespaces will also be ordered alphabetically.
+
+Given:
+```cpp
+namespace foo {
+  struct i;
+}
+
+namespace bar {
+  struct i;
+}
+```
+
+The order of the two structs w/ type `i` types would be `bar::i < foo::i`
+
+### anonymous namespace
+The anonymous namespace will be ordered after the global namespace but before
```

Comment:
an anonymous namespace shall be ordered after its enclosing namespace but before any other namespaces.